### PR TITLE
Failure handler

### PR DESCRIPTION
**Pull request by brendandeere**
In order to make sure that installments are always marked for
reprocessing, ensure that even if an error is raised any installments
not associated to a completed order are marked as failed so that they
will be reprocessed in the future

### CODE REVIEW

File: app/models/solidus_subscriptions/failure_dispatcher.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    # Run after failure callback methods
+    def dispatch
+      installments.each(&:failed)
+      notify
+    end
+
+    private
+
+    def notify
+      # Tell somone stuff happened
```
Comment (alexblackie) [inline app/models/solidus_subscriptions/failure_dispatcher.rb]: >someone

File: spec/models/solidus_subscriptions/failure_dispatcher_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe SolidusSubscriptions::FailureDispatcher do
+  let(:dispatcher) { described_class.new(installments) }
+  let(:installments) { build_list(:installment, 2) }
+
+  describe 'initialization' do
+    subject { dispatcher }
+    it { is_expected.to be_a described_class }
```
Comment (alexblackie) [inline spec/models/solidus_subscriptions/failure_dispatcher_spec.rb]: This isn't really testing anything... `described_class.new.is_a? described_class`

Comment (brendandeere) [inline spec/models/solidus_subscriptions/failure_dispatcher_spec.rb]: Yeah, youre right. I think I fell into this pattern by creating empty classes that only had an initualize method, then the code coverage would drop.... I can remove it now

**Comment by brendandeere** [comment]
@alexblackie comments addressed :)

File: app/models/solidus_subscriptions/failure_dispatcher.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# A handler for behaviour that should happen after installments are marked as
+# failures
+module SolidusSubscriptions
+  class FailureDispatcher
+    attr_reader :installments
+
+    # Get a new instance of the FailureDispatcher
+    #
+    # @param [Array<SolidusSubscriptions::Installment>] :installments,
```
Comment (alexblackie) [inline app/models/solidus_subscriptions/failure_dispatcher.rb]: wouldn't the YARD for this be:
```ruby
@param installments [Array<SolidusSubscriptions::Installment>] ...
```

Comment (brendandeere) [inline app/models/solidus_subscriptions/failure_dispatcher.rb]: Shoot.... youre right.... Ive been doing it wrong this whole time....

Comment (brendandeere) [inline app/models/solidus_subscriptions/failure_dispatcher.rb]: https://www.pivotaltracker.com/story/show/130848823

**Pull request merged**